src/OctoshiftCLI.Tests/GithubApiTests.cs

```diff
@@ -877,6 +877,26 @@ public async Task StartMigration_Returns_New_Repository_Migration_Id()
             expectedRepositoryMigrationId.Should().Be(actualRepositoryMigrationId);
         }
 
+        [Fact]
+        public async Task GetArchiveMigrationStatus_Retries_On_Failure_Status()
+        {
+            // Arrange
+            var url = $"https://api.github.com/orgs/{GITHUB_ORG}/migrations/1";
+
+            var failedResponse = "{\"state\": \"failed\"}";
+            var successfulResponse = "{\"state\": \"exported\"}";
+
+            _githubClientMock.SetupSequence(x => x.GetAsync(url, null).Result)
+                .Returns(failedResponse)
+                .Returns(failedResponse)
+                .Returns(successfulResponse);
+            // Act
+            _ = await _githubApi.GetArchiveMigrationStatus(GITHUB_ORG, 1);
+
+            // Assert
+            _githubClientMock.Verify(x => x.GetAsync(url, null), Times.Exactly(3));
```

Comment:
I would also assert than we're getting the expected response:
```suggestion
            
            // Act
            var migrationStatus = await _githubApi.GetArchiveMigrationStatus(GITHUB_ORG, 1);

            // Assert
            migrationStatus.Should().Be("exported");
            _githubClientMock.Verify(x => x.GetAsync(url, null), Times.Exactly(3));
```

Reply:
Resolved in 2e3d1a26dd8b0241b6f6e204181c0af5058c5ffd :tada: